### ISSUE TITLE
Displaying information provided by Gen 7's in-game battle UI

### ISSUE THREAD

**Issue by MacChaeger** [Zarel/Pokemon-Showdown-Client#890]
In Sun & Moon, the battle interface provides the player with quite a bit more information than it did last gen, and has actually outdone PS's own client in that regard. Information provided to the player in-game that the client doesn't display includes (but is not necessarily limited to) the following.

- [ ] The effectiveness of attacking moves on active opponents is displayed to the player as `Super effective`, `Not very effective`, `Effective`, or `No effect` when they are selecting their moves. This information is entirely based on typing and does not account for items, abilities, or other effects except Soak, Protean, etc., and Illusion (but more on that later). If a move is doubly or triply super effective or resisted, they will appear the same as moves that are only singly so. For example, Earthquake appears as `Super Effective` against Tyranitar, Eelektross, and Magnezone. Moves that deal a fixed amount of damage, such as Night Shade, will not have any information supplied about their effectiveness, even against targets that are immune. In doubles, the move itself will display the highest effectiveness possible between the two opponents, but when selecting a target, individual effectiveness for each possible target is displayed.
- [ ] The actual type of Arceus and Silvally that have been hacked to appear as a different type than they really are is revealed to both players through the game's interface. However, a Pokémon disguised through Illusion will appear to be the actual type of the Pokémon it's disguised as, even so far as to not display type changes made through the likes of Soak.
- [ ] The number of turns remaining until Future Sight and Doom Desire hit are telegraphed to the players as a status effect on the target, similar to how Perish Song works now.

If there's any other information that the in-game UI displays that PS doesn't, tell me and I'll add it.

**Comment by Zarel** [comment]
Yeah, I want to implement this eventually. I want to take abilities/items into account, though, which is putting this off.

**Comment by Marty-D** [comment]
If you Transform into an opposing Pokemon, the Transform user's Ability will be displayed as the actual Ability of the target (the summary screen still shows your original Ability as in every other gen). Obviously, if it's not possible to check the battle screen before you get forced out or knocked out then you don't learn anything, so it would be an information leak if we updated the mouseover instantly.

**Comment by MacChaeger** [comment]
@Marty-D I assume it's the same with Imposter?

**Comment by MacChaeger** [comment]
For Doom Desire and Future Sight, would it be better to add a volatile with a duration à la Perish Song to the initial target of the move as well as any pokemon that enters that field position, or would it be better to use the same system as Reflect and other side conditions?

**Comment by Zarel** [comment]
Perish Song has manual countdown.

Doom Desire and Future Sight should probably be implemented as side conditions.
